### PR TITLE
Replace deprected/removed C++98 function adapters by inline type defi…

### PR DESCRIPTION
**Pull request by DanielaE**
…nitions.

Signed-off-by: Daniela Engert <dani@ngrt.de>

### CODE REVIEW

File: test/test.hpp [hunk 1]
```diff
@@ -127,8 +127,10 @@ struct xpr_test_case
 // test_runner
 template<typename BidiIter>
 struct test_runner
-  : std::unary_function<xpr_test_case<BidiIter>, void>
 {
+    typedef BidiIter argument_type;
```
Comment (ericniebler) [inline test/test.hpp]: This doesn't look right to me.

Comment (DanielaE) [inline test/test.hpp]: You're so right. I hate this copy-and-paste thing so much: neither proof-reading nor any of my compilers spotted this in any of the testsuite runs.

Fixed and passes testsuite on msvc 9.0, 12.0, and 14.1

**Comment by Lastique** [comment]
Thanks Eric for merging this PR. Could you also merge it to master, please?

**Comment by ericniebler** [comment]
> Thanks Eric for merging this PR. Could you also merge it to master, please?

Done.

**Pull request merged**